src/deadline/job_attachments/upload.py

```diff
@@ -906,10 +906,11 @@ def _find_matched_root_from_local_type_locations(
                 )
             return matched_root
         else:
+            keys_upper = [key.upper() for key in groupings.keys()]
```

Comment:
Could we use `os.path.normcase` (see https://docs.python.org/3/library/os.path.html#os.path.normcase) instead of upper for this purpose? With that, the same code can have correct OS-specific behavior.

Reply:
I fixed it to use `os.path.normcase` instead of `upper`. I also added a test case for non-Windows.